### PR TITLE
Studio UI Roslyn Implementation + Validation

### PR DESCRIPTION
**Pull request by openbots-ff**

### CODE REVIEW

File: OpenBots.Studio/UI/CustomControls/CommandControls.cs [hunk 1]
```diff
@@ -1381,7 +1365,7 @@ public IfrmAdvancedUIElementRecorder CreateAdvancedUIElementRecorderForm()
         public IfrmCommandEditor CreateCommandEditorForm(List<AutomationCommand> commands, List<ScriptCommand> existingCommands)
         {
             frmCommandEditor editor = new frmCommandEditor(commands, existingCommands, null);
-            editor.ScriptEngineContext.Container = _container;
+            //editor.ScriptEngineContext.Container = _container;
```
Comment (openbots-ff) [inline OpenBots.Studio/UI/CustomControls/CommandControls.cs:1368]:
```suggestion
```

File: OpenBots.Studio/UI/Forms/ScriptBuilder Forms/frmScriptBuilderListView.cs [hunk 2]
```diff
@@ -444,17 +433,17 @@ private void LoadSequenceCommand(ListViewItem selectedCommandItem, ScriptCommand
...
 
                 //update variables/elements/arguments
-                _scriptVariables = newSequence.ScriptVariables.Where(x => !string.IsNullOrEmpty(x.VariableName)).ToList();
-                _scriptElements = newSequence.ScriptElements.Where(x => !string.IsNullOrEmpty(x.ElementName)).ToList();
-                _scriptArguments = newSequence.ScriptArguments.Where(x => !string.IsNullOrEmpty(x.ArgumentName)).ToList();
-                _importedNamespaces = newSequence.ImportedNamespaces;
+                //_scriptContext.Variables = newSequence.ScriptVariables.Where(x => !string.IsNullOrEmpty(x.VariableName)).ToList();
+                //_scriptContext.Elements = newSequence.ScriptElements.Where(x => !string.IsNullOrEmpty(x.ElementName)).ToList();
+                //_scriptContext.Arguments = newSequence.ScriptArguments.Where(x => !string.IsNullOrEmpty(x.ArgumentName)).ToList();
+                //_scriptContext.ImportedNamespaces = newSequence.ImportedNamespaces;
```
Comment (openbots-ab) [inline OpenBots.Studio/UI/Forms/ScriptBuilder Forms/frmScriptBuilderListView.cs:439]: Please remove these commented lines if not required

File: OpenBots.Studio/UI/Forms/Sequence Forms/frmSequenceVariableArgument.cs [hunk 3]
```diff
@@ -36,16 +36,21 @@ private void dgvVariablesArguments_CellEnter(object sender, DataGridViewCellEven
...
 
                 //variable/argument name column
                 if (e.ColumnIndex == 0)
                 {
-                    var cellValue = dgv.Rows[e.RowIndex].Cells[0].Value;
+                    var cellValue = nameCell.Value;
+
+                    CodeDomProvider provider = CodeDomProvider.CreateProvider("C#");
 
                     CodeDomProvider provider = CodeDomProvider.CreateProvider("C#");
```
Comment (openbots-ab) [inline OpenBots.Studio/UI/Forms/Sequence Forms/frmSequenceVariableArgument.cs:55]: Please remove line 55 as provider is already defined.

**Review by openbots-ab — CHANGES_REQUESTED** [review]

File: OpenBots.Studio/UI/Forms/Sequence Forms/frmSequenceVariableArgument.cs (hunk 3 above)
Comment (openbots-ff) [reply OpenBots.Studio/UI/Forms/Sequence Forms/frmSequenceVariableArgument.cs]: done

**Review by openbots-ab — APPROVED** [review]

**Review by openbots-ch — APPROVED** [review]

**Pull request merged**